src/main/java/com/google/sps/servlets/ElectionServlet.java

```diff
@@ -14,160 +14,51 @@
 import java.util.ArrayList;
 import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import java.lang.*;
-import java.util.logging.Logger;
-import java.util.logging.Level;
-
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.io.InputStreamReader;
-import java.io.UnsupportedEncodingException;
-import java.net.HttpURLConnection;
-import java.net.MalformedURLException;
-import java.net.URL;
-import java.net.URLEncoder;
-
-import com.google.gson.Gson;
-import com.google.gson.GsonBuilder;
-import com.google.gson.JsonSyntaxException;
-import com.google.appengine.api.datastore.DatastoreService;
-import com.google.appengine.api.datastore.DatastoreServiceFactory;
-import com.google.appengine.api.datastore.Entity;
-import com.google.appengine.api.datastore.FetchOptions;
-import com.google.sps.data.Election;
-import com.google.appengine.api.datastore.PreparedQuery;
-import com.google.appengine.api.datastore.Query;
-import com.google.appengine.api.datastore.Query.SortDirection;
-import com.google.appengine.api.datastore.Key;
-import com.google.appengine.api.datastore.KeyFactory;
-import org.json.JSONArray;
-import org.json.JSONException;
-import org.json.JSONObject;
 
 @WebServlet("/election")
 
 /**
- * This servlet is used to update the gVote database with information on the ongoing elections that
+ * This servlet is used to retrieve the information on the ongoing elections that
  * an eligible voter can participate in on a given day.
  */
 public class ElectionServlet extends HttpServlet {
-
-  private static final String baseURL = "https://www.googleapis.com/civicinfo/v2/elections?key=";
-  private static final Logger logger = Logger.getLogger(ElectionServlet.class.getName());
-
   @Override
-  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
-
-        // Deleting the queries from yesterday in the case that they are outdated
-        Query query = new Query("Election");
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        PreparedQuery results = datastore.prepare(query);
-
-        for (Entity entity : results.asIterable()) {
-            Key electionEntityKey = KeyFactory.createKey("Election", entity.getKey().getId());
-            datastore.delete(electionEntityKey);
-        }
-
-        StringBuilder strBuf = new StringBuilder();  
-        HttpURLConnection conn = null;
-        BufferedReader reader = null;
-
-        try {  
-            // Requires storing API key in environment variable.
-            URL url = new URL(baseURL + System.getenv("GOOGLE_API_KEY"));
-            conn = (HttpURLConnection) url.openConnection();  
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Election");
```

Comment:
Is there a "builder" version of query? If so, please prefer that. Builders are nice because they document variables and help prevent bugs related to variable mis-ordering.

Reply:
Doing so would require using the cloud version of Datastore (https://cloud.google.com/datastore/docs/concepts/queries) rather than the appengine version we import. This would be primarily a design choice, but would also mean changing a lot of our code because we have been assuming use of the appengine version, which is what we were taught in the tutorials.